openapi-generator/src/main/kotlin/org/jellyfin/openapi/Generator.kt

```diff
@@ -39,26 +43,50 @@ class Generator(
 	private fun createApiConstants(info: Info): FileSpec = openApiConstantsBuilder.build(info)
 		.let(fileSpecBuilder::build)
 
-	fun generate(
-		openApiJson: String,
-		apiOutputDir: File,
-		modelsOutputDir: File,
-	) {
+	private fun generateInternal(openApiJson: String): GeneratorResult {
 		val parseResult = parse(openApiJson)
 
 		// Get relevant OpenAPI parts
 		val schemas = parseResult.openAPI.components.schemas
 		val paths = parseResult.openAPI.paths
 
+		// Create models
+		val models = createModels(schemas)
+		// Create API operations
+		val apis = createApis(paths)
+		// Create API constants
+		val constants = createApiConstants(parseResult.openAPI.info)
+
+		return GeneratorResult(models, apis, constants)
+	}
+
+	fun verify(
+		openApiJson: String,
+		apiOutputDir: File,
+		modelsOutputDir: File,
+	): Boolean {
+		val verification = Verification(apiOutputDir, modelsOutputDir)
+		val result = generateInternal(openApiJson)
+
+		return verification.verify(result)
```

Comment:
Would work as well ig. Creating the `Verification` first when it isn't needed yet is a bit confusing.
```suggestion
		val result = generateInternal(openApiJson)

		return Verification(apiOutputDir, modelsOutputDir).verify(result)
```

Reply:
I did it this way for when I (maybe) add checks to the `init{}` of the Verification class. Stuff like checking if the supplied dirs exist. This would create a fast-fail behavior.

Reply:
I see. That makes sense, nvm then.